doc/manpages/faq.asciidoc

```diff
@@ -0,0 +1,123 @@
+
+Example: `%|indent<ret>` to indent an entire buffer with C code.
+
+Note that some languages have a default formatter set, which you can use
+with the `:format` command.
+
+Can Kakoune automatically complete the parameters of my functions ?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+As mentioned in the above question about Vim's `=` key, Kakoune
+does not parse the contents of a buffer by itself, which makes it
+impossible for the editor to propose candidates upon completion.
+
+However, support for such a feature can be achieved through the
+use of a dedicated tool, as is the case with `clang` and C code:
+you can use the `clang-enable-autocomplete` and `clang-complete`
+builtin commands whenever editing a C/C++ file, and completion will
+work on function parameters.
+
+Note that the same features are available for python buffers, with
+the `jedi` script.
+
+Why aren't widely known command line shortcuts such as <c-w> or <c-u> available in Kakoune ?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Despite their widespread availability in multiple tools, those
+shortcuts do not fit the paradigm that Kakoune implements, which is
+based on selections first.
+
+However, you can easily declare key mappings in your configuration
+file to be able to use those control-based shortcuts in insert mode
+(c.f. the "map" command in the "commands" documentation page).
+
+How can I explore the filesystem the way Vim's NerdTree does ?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The builtin file completion engine used when opening a file for editing
+(using the `:edit` command and letting the suggestions popup in the
+menu beneath) is much more convenient than Vim's, which should fill
+basic needs.
+
+However, if you need an actual explorer to interact with the editor,
+you can create a Kakoune script that will spawn the tool in question,
+which should in return send an "edit" command followed by the path
+of the file you selected to the current Kakoune session (e.g. `echo
+eval -client $kak_client edit /path/to/file | kak -p $kak_session`).
+
+Why aren't there other scopes except for `%sh{}` e.g. python ?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Supporting custom scopes would add hard dependencies to the project,
+which is too much of an drawback when balanced against the low cost of
+using an interpreter in a regular shell scope (e.g. `%sh{ python -c
+"..." }`), which has a negligible impact on performance.
+
+Can I disable auto-indentation completely ?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+All the indentation hooks are conventionally named `<lang>-indent`, which
+allows us to use the `disabled_hooks` variable to disable indentation globally
+with the following command: `set global disabled_hooks '.+-indent'`
+
+Can I disable syntax highligting completely ?
```

Comment:
And thanks to completion, this typo has been inserted two more times afterwards :(

Thanks for noticing.